lib/ansible/modules/rpm_key.py

```diff
@@ -194,47 +205,58 @@ def getfingerprint(self, keyfile):
                 #
                 # "fpr :: Fingerprint (fingerprint is in field 10)"
                 #
-                return line.split(':')[9]
+                fingerprints.append(line.split(':')[9])
+
+        if fingerprints == []:
+            self.module.fail_json(msg="Unexpected gpg output")
 
-        self.module.fail_json(msg="Unexpected gpg output")
+        return fingerprints
 
-    def is_keyid(self, keystr):
-        """Verifies if a key, as provided by the user is a keyid"""
-        return re.match('(0x)?[0-9a-f]{8}', keystr, flags=re.IGNORECASE)
+    def is_keyid(self, keys):
+        """Verifies if keys, as provided by the user are keyids"""
+        for keystr in keys:
+            return re.match('(0x)?[0-9a-f]{8}', keystr, flags=re.IGNORECASE)
 
     def execute_command(self, cmd):
         rc, stdout, stderr = self.module.run_command(cmd, use_unsafe_shell=True)
         if rc != 0:
             self.module.fail_json(msg=stderr)
         return stdout, stderr
 
-    def is_key_imported(self, keyid):
+    def is_key_imported(self, keyids):
         cmd = self.rpm + ' -q  gpg-pubkey'
         rc, stdout, stderr = self.module.run_command(cmd)
         if rc != 0:  # No key is installed on system
             return False
         cmd += ' --qf "%{description}" | ' + self.gpg + ' --no-tty --batch --with-colons --fixed-list-mode -'
         stdout, stderr = self.execute_command(cmd)
+
+        matching_keys = 0
         for line in stdout.splitlines():
-            if keyid in line.split(':')[4]:
+            if matching_keys == len(keyids):
                 return True
+            for keyid in keyids:
+                if keyid in line.split(':')[4]:
+                    matching_keys += 1
```

Comment:
We're concerned about the potential edge cases where this may return a false positive. I think we'd be more comfortable, if we actually tracked the matching keys, and compared them against the actual content of `keyids` instead of asserting the lengths are equal.

Reply:
Good point! I'll update this so it does an actual comparison.